packages/core/js-client/src/particle/Particle.ts

```diff
@@ -67,11 +88,19 @@ export const hasExpired = (particle: IParticle): boolean => {
     return getActualTTL(particle) <= 0;
 };
 
+/**
+ * Validates that particle signature is correct
+ */
+export const verifySignature = async (particle: IParticle, publicKey: PublicKey): Promise<boolean> => {
+    const message = buildParticleMessage(particle);
```

Comment:
You accept `IParticle` which doesn't have any fields except `data`.

But `buildParticleMessage` expects more fields `{ id, initPeerId, timestamp, ttl, script }`.

How does it typecheck?

Reply:
IParticle extends another interface with all immutable fiendls